### PR TITLE
[BE]: Update mypy to 1.10.0

### PR DESCRIPTION
**Pull request by Skylion007**
Updates mypy to the latest and greatest.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/distributed/_composable/fsdp/fully_shard.py [hunk 1]
```diff
@@ -345,4 +345,8 @@ def wrapped_method(self, *args, **kwargs):
         return fsdp_state._post_forward(self, args, out)
 
     # Use `__get__` to make `wrapped_method` an instance method
-    setattr(module, method_name, wrapped_method.__get__(module, type(module)))
+    setattr(
+        module,
+        method_name,
+        wrapped_method.__get__(module, type(module)),  # type:ignore[attr-defined]
```
Comment (awgu) [inline torch/distributed/_composable/fsdp/fully_shard.py:351]: @Skylion007 Do you know why `__get__` is not always defined?

**Pull request closed without merging**